### ISSUE TITLE
"he_heavy_augment" model

### ISSUE THREAD

**Issue by afrendeiro** [stardist/stardist-imagej#9]
Hello,

Thank you so much for stardist and above all making access to pre-trained models so easy - 100% fan.

I've used the ["he_heavy_augment" model](https://github.com/stardist/stardist-imagej/blob/master/src/main/resources/models/2D/he_heavy_augment.zip) with great success using the ImageJ stardist plugin (will cite stardist of course).

However, I'd like to run this model outside ImageJ using the stardist Python package, but it isn't available in the stardist repository. Maybe there is a trivial way to load/convert the keras model into a stardist model, but I have not yet managed.
Any help/pointers would be appreciated.

Cheers, Andre

**Comment by maweigert** [comment]
Hi @afrendeiro ,

Thanks a lot for your feedback - highly appreciated 😄 

> However, I'd like to run this model outside ImageJ using the stardist Python package, but it isn't available in the stardist repository.

Yes, you can load the pretrained models as described in the [python readme](https://github.com/mpicbg-csbd/stardist#pretrained-models-for-2d):

```
from stardist.models import StarDist2D 

model = StarDist2D.from_pretrained('2D_versatile_he')

model.predict_instances(img, ...) 
```

Hope that helps!

Martin

**Comment by afrendeiro** [comment]
Hmm, I did try that model - I guess my confusion comes from having very different results with the two implementations:

![impl_comparison](https://i.ibb.co/X8wpKyf/Screenshot-from-2021-01-12-11-02-26.png)

I assumed the models were different since the names are also slightly different.

I guess this comes from having different default arguments?

**Comment by maweigert** [comment]
Hi, 

The defaults should be the same. Can you post the image that you are using?

**Comment by afrendeiro** [comment]
Yes, here it is: https://wcm.box.com/shared/static/9mqe6bc1wm04f8cur8kpbzdlvs802yep.tif

And a little script to reproduce:

```python
import io
import tempfile
import subprocess
from pathlib import Path

import requests
import numpy as np
import tifffile
import matplotlib.pyplot as plt
from stardist import random_label_cmap
from stardist.models import StarDist2D

IMAGE_J_PATH = Path("~/Downloads/fiji/Fiji.app/ImageJ-linux64").expanduser().absolute()
IMAGE_URL = 'https://wcm.box.com/shared/static/9mqe6bc1wm04f8cur8kpbzdlvs802yep.tif'

def get_image_from_url(url: str) -> np.ndarray:
    req = requests.get(url)
    if not req.ok:
        raise ValueError
    buf = io.BytesIO()
    buf.write(req.content)
    buf.seek(0)
    return tifffile.imread(buf)

def segment_imagej(input_file: Path, output_file: Path) -> None:
    macro_file = tempfile.NamedTemporaryFile(suffix='.py').name

    macro = f"""# @ DatasetIOService io
# @ CommandService command

from de.csbdresden.stardist import StarDist2D

f = "{input_file}"
imp = io.open(f)

res = command.run(
    StarDist2D,
    False,
    "input",
    imp,
    "modelChoice",
    "Versatile (H&E nuclei)",
).get()
label = res.getOutput("label")

io.save(label, "{output_file}")
    """
    with open(macro_file, 'w') as handle:
        handle.write(macro)
    # !subl $macro_file

    cmd = f"{IMAGE_J_PATH} --ij2 --headless --console --run {macro_file}"
    o = subprocess.call(cmd.split(" "))


_dir = Path(".").absolute()
img_file = _dir / 'test_image.tiff'
maskij_file = _dir / 'test_mask.imj.tiff'
maskpy_file = _dir / 'test_mask.py.tiff'

# Get and save image
img = get_image_from_url(IMAGE_URL)
tifffile.imwrite(img_file, img)

# Segment with ImageJ
segment_imagej(img_file, maskij_file)
maskij = tifffile.imread(maskij_file)

# Segment with native
model = StarDist2D.from_pretrained('2D_versatile_he')
maskpy, prob = model.predict_instances(img)
tifffile.imwrite(maskpy_file, maskpy)


# Plot
fig, axes = plt.subplots(1, 3, figsize=(3 * 4, 4))
axes[0].imshow(img)
axes[0].set(title="Original")
axes[0].axis('off')
for ax, mask, title in zip(axes[1:], [maskij, maskpy], ['ImageJ', "Python"]):
    # mask = np.ma.masked_array(mask, mask==0)
    ax.imshow(mask, cmap=random_label_cmap())
    ax.axis('off')
    ax.set(title=title)
fig.savefig(_dir / "test_result.png", dpi=300, bbox_inches="tight")
```
One'd need to change `IMAGE_J_PATH`.

I didn't quite get it to run as is for an unrelated issue (for some reason the macro won't save the image when run with headless ImageJ, but works fine if I run it in ImageJ console ¯\_(ツ)_/¯

**Comment by maweigert** [comment]
Ah, I see. I suspect the image was not normalized before predicting in python (which is automatically done in Fiji)?

The following does work for me: 

```python 

from stardist.models import StarDist2D
from csbdeep.utils import normalize

x = imread(...)

model = StarDist2D.from_pretrained('2D_versatile_he')

labels, _ = model.predict_instances(normalize(x,1,99.8))
```

**Comment by afrendeiro** [comment]
Ah duh - that explains it. Thanks a lot and sorry about bothering you!
